### ISSUE TITLE
Can't connect to my maria DB

### ISSUE THREAD

**Issue by Shasoosh** [linuxserver/docker-kodi-headless#122]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]
	

I have 4 instances of Kodi 18.8 on Android and on Windows, all working with a populated MariaDB sitting on my Synology NAS
I've edited my advancedsettings.xml with the same params as my instances of kodi advancedsettings.xml
```
<videodatabase>
<type>mysql</type>
<!-- don't change this value -->
<host>192.168.1.10</host>
<!-- change this to the IP address of your mysql/mariadb server  -->
<port>3307</port>
<!-- change this to the port of your mysql/mariadb server  -->
<user>kodi</user>
<!-- change this to the user for your database, not root  -->
<pass>*****</pass>
<!-- change this to the password for your database -->
</videodatabase>
```

Inside the kodi.log I see this with no errors: 
`2020-11-05 14:43:31.655 T:140582243493632  NOTICE: MYSQL: Connected to version 5.5.5-10.3.21-MariaDB`

But when I'm accessing the kodi-headless interface via http://192.168.1.10:8070, clicking on 'scan video library' I'm getting 'Lost websocket connection' every time and my media isn't accessible from there.

```
2020-11-05 16:43:47.136 T:139987370616576  NOTICE: ES: Starting UDP Event server on port 9777
2020-11-05 16:43:47.136 T:139987370616576  NOTICE: UDP: Listening on port 9777 (ipv6 : true)
2020-11-05 16:43:48.025 T:139987656083200 WARNING: CActiveAE::StateMachine - signal: 22 from port: timer not handled for state: 1
2020-11-05 16:51:27.153 T:139987001534208 WARNING: Previous line repeats 458 times.
2020-11-05 16:51:27.154 T:139987001534208  NOTICE: VideoInfoScanner: Starting scan ..
2020-11-05 16:51:27.154 T:139987001534208  NOTICE: VideoInfoScanner: Finished scan. Scanning for video info took 00:00
2020-11-05 16:51:27.156 T:139987656083200 WARNING: CActiveAE::StateMachine - signal: 22 from port: timer not handled for state: 1
```
As you can see, at 16:51:27.153 I tried running a python script from my Windows 10 machine in order to update the video library, the screen managed to access my NAS machine and the scanning process started and ended after 0 seconds.

Is there any other log output that can help me here?
Is this related to the fact that all my instances are using Kodi 18.8 (and the mariaDb was created using Kodi 18.8) while the headless is 18.3 ?

Thanks

**Comment by j0nnymoe** [comment]
kodi-headless is designed to fit into an existing setup, not create a fresh database. You need to scan your media with a real client.

**Comment by Shasoosh** [comment]
There's an existing db on 192.168.1.10:3307 with a lot of content. I can't get to that content from headless.
Opening a real kodi client from Windows shows all content from that db.

**Comment by j0nnymoe** [comment]
In theory you should be able to browse the content is there and will receive update notifications.
Using kodi in a headless state is hacky at best as it's not designed to run like this. A better solution is just to use something like Plex/Emby/JellyFin as a media backend for your kodi clients.

**Comment by Shasoosh** [comment]
I think I'm miss understood. My main intention is to update Kodi remotely. I don't care if I can't browse content from headless.
It's just one of my indication (that and the websocket error) that something is wrong.

Obviously trying to update the video library remotely also fails.

**Comment by j0nnymoe** [comment]
Have you copied over your sources.xml?

**Comment by Shasoosh** [comment]
Yes. 

My sources.xml file sources are in this format:

```
        <source>
            <name>Kids Movies</name>
            <path pathversion="1">nfs://192.168.1.10/volume1/Media/Videos/Movies/Kids/</path>
            <allowsharing>true</allowsharing>
        </source>
```

Is the nfs:// a valid prefix to be read from within the NAS intself?

**Comment by hugepants** [comment]
> Is there any other log output that can help me here?

Ensure that this line is present: `NOTICE: Running database version MyVideos116`

Then enable debug logging by adding the following line in `advancedsettings.xml` and look for errors:

```
<loglevel>2</loglevel>
```

> Is this related to the fact that all my instances are using Kodi 18.8 (and the mariaDb was created using Kodi 18.8) while the headless is 18.3 ?

No, the db schema does not usually change in point releases.


> Is the nfs:// a valid prefix to be read from within the NAS intself?

Depends how the export is configured but you can try a path substitution in `advancedsettings.xml`:

```
<pathsubstitution>
  <substitute>
    <from>nfs://192.168.1.10/volume1/</from>
    <to>/volume1/</to>
  </substitute>
</pathsubstitution>
```

And map `/volume1` to `/volume1` in the container
This assumes `/volume1` is the original path, amend if necessary.

**Comment by Shasoosh** [comment]
Thanks for the reply. I've removed the sources.xml file from my userdata folder to avoid the extra noise as my current problem is with scanning the library and to my understanding, sources.xml isn't needed for that.

> Ensure that this line is present: `NOTICE: Running database version MyVideos116`

This line is present. I've enabled loglevel 2 and pasted it here: https://pastebin.com/aNJeR4DJ

I don't see any errors. It's like Kodi is able to access the db but thinks it's empty.

**Comment by hugepants** [comment]
The `sources.xml` is needed. The database side appears to be OK so it looks like the library scan is failing because Kodi can't access the media sources. You need to either:

1) Ensure your NFS export configuration allows for localhost/loopback mounting
or
2) Use path substitution as described.

With the [texturecache.py](https://github.com/MilhouseVH/texturecache.py) tool, run `texturecache.py sources` to list the sources visible to Kodi.

**Comment by Shasoosh** [comment]
All my NFS permissions are enabled as far as I can see.

Running texturecache I get this:

```
python texturecache.py sources
video: Kids Movies: nfs://192.168.1.10/volume1/Media/Videos/Movies/Kids
video: Kids TV: nfs://192.168.1.10/volume1/Media/Videos/TV/Kids
video: Kids-4K: nfs://192.168.1.10/volume1/Media/Videos/Movies/Kids-4K
video: Sha Movies: nfs://192.168.1.10/volume1/Media/Videos/Movies/Sha
video: Sha Tv: nfs://192.168.1.10/volume1/Media/Videos/TV/Sha
files: Media: nfs://192.168.1.10/volume1/Media/Videos
```

If I go to the headless UI and click on Browser, under Video I see all my sources. Clicking on one of them shows me all the movie folders inside and the log shows this:

```
2020-11-06 13:41:24.987 T:139746575619840   DEBUG: CUtil::GetMatchingSource: no matching source found for [nfs://192.168.1.10/volume1/Media/Videos/Movies/Kids/]
2020-11-06 13:41:24.989 T:139746575619840   DEBUG: NFS: Context for 192.168.1.10/volume1/Media not open - get a new context.
2020-11-06 13:41:24.990 T:139746617583360   DEBUG: CWebServer[8080]: request received for /jsonrpc?Player.GetActivePlayers
2020-11-06 13:41:24.990 T:139746575619840   DEBUG: NFS: Connected to server 192.168.1.10 and export /volume1/Media
2020-11-06 13:41:24.990 T:139746575619840   DEBUG: NFS: chunks: r/w 131072/131072
2020-11-06 13:41:25.005 T:139746584012544   DEBUG: JSONRPC Server: New connection detected
2020-11-06 13:41:25.006 T:139746584012544    INFO: JSONRPC Server: New connection added
2020-11-06 13:41:25.006 T:139746617583360   DEBUG: CWebServer[8080]: request received for /jsonrpc?PlaylistCollection
2020-11-06 13:41:25.025 T:139746617583360   DEBUG: CWebServer[8080]: request received for /icon.png
2020-11-06 13:41:25.579 T:139747014833920 WARNING: CActiveAE::StateMachine - signal: 22 from port: timer not handled for state: 1
```
Yet, running a scan still does nothing.

**Comment by Shasoosh** [comment]
Other things I've tried:
Re install MariaDB as a container and not as a package through my Synology NAS.
Re install MariaDB version 5.5.64 instead of the latest 10.*
Nuked everything and went with an SMB route.
Populated my DB, edited the `advancedsettings.xml` on my headless and run it. Still no go. Trying to click on the Video source inside the headless UI browser, I got SMB access errors in the kodi log so I've added this:

```
  <pathsubstitution>
    <substitute>
      <from>smb://192.168.1.10/Media</from>
      <to>/kodimedia</to>
    </substitute>
  </pathsubstitution>
```

Now logging into the headless UI and going to the browser button, I see my source and able to click on it and see a list of movies (so no permission problem there) BUT i'm still getting `Lost websocket connection` when trying to scan my video library.
When trying to scan it seems like kodi isn't even trying to access my sources. Always takes 0 seconds with no errors.

I'm really stumped at this point
